protocol/v2/ssv/runner/aggregator_committee.go

```diff
@@ -0,0 +1,1460 @@
+	if err := json.Unmarshal(data, &aux); err != nil {
+		return err
+	}
+
+	// Assign fields
+	r.BaseRunner = aux.BaseRunner
+	r.beacon = aux.beacon
+	r.network = aux.network
+	r.signer = aux.signer
+	r.operatorSigner = aux.operatorSigner
+	r.ValCheck = aux.valCheck
+	return nil
+}
+func (r *AggregatorCommitteeRunner) HasRunningQBFTInstance() bool {
+	return r.BaseRunner.HasRunningQBFTInstance()
+}
+
+func (r *AggregatorCommitteeRunner) HasAcceptedProposalForCurrentRound() bool {
+	return r.BaseRunner.HasAcceptedProposalForCurrentRound()
+}
+
+func (r *AggregatorCommitteeRunner) GetShares() map[phase0.ValidatorIndex]*spectypes.Share {
+	return r.BaseRunner.GetShares()
+}
+
+func (r *AggregatorCommitteeRunner) GetRole() spectypes.RunnerRole {
+	return r.BaseRunner.GetRole()
+}
+
+func (r *AggregatorCommitteeRunner) GetLastHeight() specqbft.Height {
+	return r.BaseRunner.GetLastHeight()
+}
+
+func (r *AggregatorCommitteeRunner) GetLastRound() specqbft.Round {
+	return r.BaseRunner.GetLastRound()
+}
+
+func (r *AggregatorCommitteeRunner) GetStateRoot() ([32]byte, error) {
+	return r.BaseRunner.GetStateRoot()
+}
+
+func (r *AggregatorCommitteeRunner) SetTimeoutFunc(fn TimeoutF) {
+	r.BaseRunner.SetTimeoutFunc(fn)
+}
+
+func (r *AggregatorCommitteeRunner) GetBeaconNode() beacon.BeaconNode {
+	return r.beacon
+}
+
+func (r *AggregatorCommitteeRunner) GetNetwork() specqbft.Network {
+	return r.network
+}
+
+func (r *AggregatorCommitteeRunner) GetNetworkConfig() *networkconfig.Network {
+	return r.BaseRunner.NetworkConfig
+}
+
+func (r *AggregatorCommitteeRunner) GetBeaconSigner() ekm.BeaconSigner {
+	return r.signer
+}
+
+func (r *AggregatorCommitteeRunner) HasRunningDuty() bool {
+	return r.BaseRunner.hasRunningDuty()
+}
+
+func (r *AggregatorCommitteeRunner) GetBaseRunner() *BaseRunner {
+	return r.BaseRunner
+}
+
+// findValidatorDuty finds the validator duty for a specific role
+func (r *AggregatorCommitteeRunner) findValidatorDuty(duty *spectypes.AggregatorCommitteeDuty, validatorIndex phase0.ValidatorIndex, role spectypes.BeaconRole) *spectypes.ValidatorDuty {
+	for _, d := range duty.ValidatorDuties {
+		if d.ValidatorIndex == validatorIndex && d.Type == role {
+			return d
+		}
+	}
+
+	return nil
+}
+
+// processAggregatorSelectionProof handles aggregator selection proofs
+func (r *AggregatorCommitteeRunner) processAggregatorSelectionProof(
+	ctx context.Context,
+	selectionProof phase0.BLSSignature,
+	vDuty *spectypes.ValidatorDuty,
+	aggregatorData *spectypes.AggregatorCommitteeConsensusData,
+) (bool, error) {
+	ctx, span := tracer.Start(ctx,
+		observability.InstrumentName(observabilityNamespace, "runner.process_aggregator_selection_proof"),
+		trace.WithAttributes(
+		// TODO
+		))
+	defer span.End()
+
+	isAggregator := r.IsAggregator(ctx, vDuty.Slot, vDuty.CommitteeIndex, vDuty.CommitteeLength, selectionProof[:])
+	if !isAggregator {
+		return false, nil
+	}
+
+	// TODO: waitToSlotTwoThirds(vDuty.Slot)
```

Comment:
we should probably turn this on. and make sure committee runner is also waiting for this or head event somewhere.